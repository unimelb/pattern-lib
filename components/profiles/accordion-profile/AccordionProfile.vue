<template>
  <div class="accordion-profile">
    <div
      :style="{ backgroundImage: `url(${thumb})` }"
      :class="thumbClass"
      class="accordion-profile__thumb"
      aria-label="Profile Image" />
    <div
      :class="containBioClass"
      class="accordion-profile__container"
      @click="changeIcon">
      <div
        :style="{ backgroundImage: `url(${thumb})` }"
        :class="smallThumbClass"
        class="accordion-profile__thumb-inner"
        aria-label="Profile Image" />
      <div class="accordion-profile__inner">
        <div class="accordion-profile__titles">
          <h6 class="accordion-profile__title">
            {{ title }}
          </h6>
          <p class="accordion-profile__sub-title">
            {{ subTitle }}
          </p>
        </div>
        <div>
          <SvgIcon
            v-if="bio"
            :class="iconClassName"
            class="accordion-profile__icon"
            name="chevron-up"
            aria-label="chevron"
            width="14px"
            height="14px" />
        </div>
      </div>
    </div>
    <div
      :class="bioClass"
      class="accordion-profile__bio">
      <VideoEmbed
        v-if="video"
        :src="video"
        :video-label="videoLabel"
        video-api
        class="accordion-profile__embed" />
      <p class="accordion-profile__bio-text">
        {{ bio }}
      </p>
    </div>
  </div>
</template>

<script>
import SvgIcon from '../../icons/SvgIcon.vue';
import VideoEmbed from '../../embed/VideoEmbed.vue';
import { IMAGE_PLACEHOLDER_SHORT } from '../../../utils/placeholders';

export default {
  name: 'Showcase',
  components: { SvgIcon, VideoEmbed },
  props: {
    thumb: {
      type: String,
      default: IMAGE_PLACEHOLDER_SHORT,
    },
    title: {
      type: String,
      default: '',
      required: true,
    },
    subTitle: {
      type: String,
      default: '',
    },
    video: {
      type: String,
      default: '',
    },
    videoLabel: {
      type: String,
      default: '',
    },
    bio: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      isHidden: true,
    };
  },
  computed: {
    thumbClass() {
      return (this.isHidden) ? '' : 'accordion-profile__thumb--transition';
    },
    smallThumbClass() {
      return (this.isHidden) ? '' : 'accordion-profile__thumb-inner--transition';
    },
    iconClassName() {
      return (this.isHidden) ? '' : 'accordion-profile__icon--active';
    },
    containBioClass() {
      return (this.bio) ? 'accordion-profile__container--bio' : '';
    },
    bioClass() {
      return (this.isHidden) ? 'accordion-profile__bio--transition' : '';
    },
  },
  methods: {
    changeIcon() {
      this.pauseVideo();
      if (this.bio !== '') {
        this.isHidden = !this.isHidden;
      }
    },
    pauseVideo() {
      const iframes = document.querySelectorAll('iframe');
      Array.prototype.forEach.call(iframes, (iframe) => {
        iframe.contentWindow.postMessage(JSON.stringify({
          event: 'command',
          func: 'pauseVideo',
        }), '*');
      });
    },
  },
};
</script>
