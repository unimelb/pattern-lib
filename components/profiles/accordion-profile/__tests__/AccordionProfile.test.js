import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import AccordionProfile from '../AccordionProfile.vue';
import {
  IMAGE_PLACEHOLDER_SHORT,
} from '../../../../utils/placeholders';


expect.extend(toHaveNoViolations);

describe('AccordionProfile', () => {
  it('should match snapshot', () => {
    const result = shallow(AccordionProfile, {
      propsData: {
        title: 'mock title',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('should have default props and correct types thumb/title/subTitle/video/bio', () => {
    const wrapper = shallow(AccordionProfile);
    const {
      thumb,
      title,
      subTitle,
      video,
      bio,
    } = wrapper.vm.$options.props;

    expect(thumb.type).toEqual(String);
    expect(wrapper.props().thumb).toBe(IMAGE_PLACEHOLDER_SHORT);

    expect(title.type).toBe(String);
    expect(wrapper.props().title).toBe('');

    expect(subTitle.type).toBe(String);
    expect(wrapper.props().subTitle).toBe('');

    expect(video.type).toBe(String);
    expect(wrapper.props().video).toBe('');

    expect(bio.type).toBe(String);
    expect(wrapper.props().bio).toBe('');
  });

  it('should render thumb from prop with correct type', () => {
    const thumb = 'http://';
    const wrapper = shallow(AccordionProfile, {
      propsData: {
        thumb: 'http://',
      },
    });

    expect(typeof wrapper.props().thumb).toBe('string');
    expect(wrapper.props().thumb).toBe(thumb);
    expect(wrapper.find('.accordion-profile__thumb').attributes().style).toBe('background-image: url(http://);');
  });

  it('should render title from prop with correct type', () => {
    const title = 'Test title';
    const wrapper = shallow(AccordionProfile, {
      propsData: {
        title: 'Test title',
      },
    });

    expect(typeof wrapper.props().title).toBe('string');
    expect(wrapper.props().title).toBe(title);
    expect(wrapper.find('h6').text()).toBe(title);
  });

  it('should render subTitle from prop with correct type', () => {
    const subTitle = 'Test subTitle';
    const wrapper = shallow(AccordionProfile, {
      propsData: {
        subTitle: 'Test subTitle',
      },
    });

    expect(typeof wrapper.props().subTitle).toBe('string');
    expect(wrapper.props().subTitle).toBe(subTitle);
    expect(wrapper.find('.accordion-profile__sub-title').text()).toBe(subTitle);
  });

  it('should render bio prop with correct type', () => {
    const bio = 'Test bio';
    const wrapper = shallow(AccordionProfile, {
      propsData: {
        bio: 'Test bio',
      },
    });

    expect(typeof wrapper.props().bio).toBe('string');
    expect(wrapper.props().bio).toBe(bio);
    expect(wrapper.find('.accordion-profile__bio').text()).toBe(bio);
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(AccordionProfile, {
      propsData: {
        title: 'mock title',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
