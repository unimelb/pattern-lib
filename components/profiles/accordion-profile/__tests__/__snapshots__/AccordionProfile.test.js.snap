// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionProfile should match snapshot 1`] = `
<div
  class="accordion-profile"
>
  <div
    aria-label="Profile Image"
    class="accordion-profile__thumb"
    style="background-image: url(https://matrix-cms.unimelb.edu.au/__data/assets/image/0028/85159/staff-placeholder-short.png);"
  />
   
  <div
    class="accordion-profile__container"
  >
    <div
      aria-label="Profile Image"
      class="accordion-profile__thumb-inner"
      style="background-image: url(https://matrix-cms.unimelb.edu.au/__data/assets/image/0028/85159/staff-placeholder-short.png);"
    />
     
    <div
      class="accordion-profile__inner"
    >
      <div
        class="accordion-profile__titles"
      >
        <h6
          class="accordion-profile__title"
        >
          
          mock title
        
        </h6>
         
        <p
          class="accordion-profile__sub-title"
        >
          
          
        
        </p>
      </div>
       
      <div>
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="accordion-profile__bio accordion-profile__bio--transition"
  >
    <!---->
     
    <p
      class="accordion-profile__bio-text"
    >
      
      
    
    </p>
  </div>
</div>
`;
